functions/src/db/getRoom.ts

```diff
@@ -0,0 +1,13 @@
+import { Room, RoomId } from '../types/Room'
+import { db } from '../firebase/firebase'
+import { NotFoundError } from '../api/errors/HttpError'
+
+export const getRoom = async (roomId: RoomId): Promise<Room> => {
+    const documentSnapshot = await db.collection('rooms').doc(roomId).get()
+
+    if (!documentSnapshot.exists) {
+        throw new NotFoundError('Resource does not exist')
+    }
+
+    return <Room>documentSnapshot.data()
```

Comment:
is type cast necessary here ? The function signature already specifies the return type

Reply:
indeed, but it is necessary, typescript doesn't get it.